### PR TITLE
CIRC-415: Change due date fails

### PR DESCRIPTION
**Pull request by VRohach**
1. Mechanism for keeping 'patronGroupIdAtCheckout' to storage if Patron object is empty in user - created.
2. Useless log regarding empty PatronGroup removed.

## Purpose

JIRA ticket: https://issues.folio.org/browse/CIRC-415

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/LoanRepresentation.java [hunk 1]
```diff
@@ -144,8 +144,6 @@ private void additionalBorrowerProperties(JsonObject loanRepresentation, User bo
```
Comment (mrdpopov) [inline src/main/java/org/folio/circulation/domain/LoanRepresentation.java]: Is this method needed?

File: src/main/java/org/folio/circulation/domain/LoanRepository.java [hunk 2]
```diff
@@ -171,12 +171,13 @@ else if(loans.getTotalRecords() == 1) {
...
     JsonObject storageLoan = loan.asJson();
 
+    keepPatronGroupIdAtCheckoutPropertiesIfLoanUpdated(storageLoan);
     removeChangeMetadata(storageLoan);
     removeSummaryProperties(storageLoan);
     keepLatestItemStatus(item, storageLoan);
     removeBorrowerProperties(storageLoan);
     removeLoanPolicyProperties(storageLoan);
-    keepPatronGroupAtCheckoutProperties(loan, storageLoan);
+    keepPatronGroupIdAtCheckoutProperties(loan, storageLoan);
```
Comment (mrdpopov) [inline src/main/java/org/folio/circulation/domain/LoanRepository.java]: Do we need two new methods? Can the be combined into one?

**Comment by marcjohnson-kint** [comment]
@VRohach @mrdpopov My understanding of the cause of the bug reported in [CIRC-415](https://issues.folio.org/browse/CIRC-415) is that `patronGroupAtCheckout` property that is part of the business logic representation of the loan is not defined in the storage representation, and so should be removed prior to storage.

This is another example of a common class of integration issues we've had with mod-circulation.

The changes in this pull request don't appear to remove this property from the storage representation. Have I missed something?

I note that this change has already been merged, so this feedback might be rather late.

cc: @zburke

**Comment by mrdpopov** [comment]
The issue was that that property has different name in mod-circulation-storage. I verified the fix locally before merging.

**Comment by marcjohnson-kint** [comment]
> @marc-johnson The issue is that that property has different name in mod-circulation-storage. I verified the fix locally before merging.

I don't follow, the storage loan representation is based upon a copy of the business logic representation (in retrospect, that was a mistake), the code change includes the `patronGroupIdAtCheckout` property but does not seem to remove the `patronGroupAtCheckout` property which is what causes the reported error.

How did you verify the fix locally?

**Comment by marcjohnson-kint** [comment]
> @marc-johnson The issue is that that property has different name in mod-circulation-storage. I verified the fix locally before merging.

Ah, I now understand, the incorrect property was already removed elsewhere.

**Comment by VRohach** [comment]
> > @marc-johnson The issue is that that property has different name in mod-circulation-storage. I verified the fix locally before merging.
> 
> Ah, I now understand, the incorrect property was already removed elsewhere.

@marc-johnson Just to describe this PR.


> @VRohach @mrdpopov My understanding of the cause of the bug reported in [CIRC-415](https://issues.folio.org/browse/CIRC-415) is that `patronGroupAtCheckout` property that is part of the business logic representation of the loan is not defined in the storage representation, and so should be removed prior to storage.
> 
> This is another example of a common class of integration issues we've had with mod-circulation.
> 
> The changes in this pull request don't appear to remove this property from the storage representation. Have I missed something?
> 
> I note that this change has already been merged, so this feedback might be rather late.
> 
> cc: @zburke

@marcjohnson-kint Yes, you are right. The "`patronGroupAtCheckout`" object should be removed from the storage representation. I did it. 

But then, I discovered, that if we update the loan, for example, via _due date_ mechanism- then **we don't have "`patronGroupIdAtCheckout`" at storage representation**, because we don't have PatronGroup in the User object in our current Loan one. But, in update case, we have "`patronGroupAtCheckout`" object with id and name in the json-representation. 

That`s why new mechanism was created for handling loan updating cases: we get patronGroup id from "`patronGroupAtCheckout`" object (with `id `and `name `fields), and create and set it to the "`patronGroupIdAtCheckout`" field for storage.

Just to describe the logic of this PR. Thanks!

**Pull request merged**